### PR TITLE
Implement iterator support at getLastKeyPair

### PR DESCRIPTION
**Pull request by cwreden**
Hello together,

we use this Guard with our Redis iterator and the Key() method from php doesn't work with iterators that have there data not as properties.
So I implement the iterator support like in enforceStorageLimit method.

I'm not sure how travis work, what are open tasks for me?

### CODE REVIEW

File: src/Guard.php [hunk 1]
```diff
@@ -289,10 +289,22 @@ protected function getLastKeyPair(): ?array
...
-        reset($this->storage);
+        $name = null;
+        $value = null;
+        if (is_array($this->storage)) {
+            end($this->storage);
+            $name = key($this->storage);
+            $value = $this->storage[$name];
+            reset($this->storage);
+        } elseif ($this->storage instanceof Iterator) {
+            $this->storage->rewind();
```
Comment (akrabat) [inline src/Guard.php:300]: I don't know if this optimisation is worth it, but if `$this->storage` implements `Countable` and `SeekableIterator`, we could do:

```php
$this->storage->seek(count($this->storage) - 1); 
$name = $this->storage->key();
$value = $this->storage->current();
```

and avoid the loop. I suppose it depends how many items are likely to be in the storage as to whether this is worth it though.

File: tests/GuardTest.php [hunk 2]
```diff
@@ -425,4 +425,25 @@ public function testProcessAppendsNewTokensWhenPersistentTokenModeIsOn()
 
         $mw->process($requestProphecy->reveal(), $requestHandlerProphecy->reveal());
     }
+
+    public function testCanGetLastKeyPairFromIterator()
+    {
+
```
Comment (akrabat) [inline tests/GuardTest.php:431]: Can you remove this blank line please?

**Review by akrabat — CHANGES_REQUESTED** [review]
This looks like a useful addition. One thing to consider and a minor whitespace update to do.

File: src/Guard.php (hunk 1 above)
Comment (cwreden) [reply src/Guard.php:300]: I think then there is a conflict with Guard.php:131. There is the Iterator interface the minimum dependency and not the SeekableIterator.

Comment (akrabat) [reply src/Guard.php:300]: Fair enough.

**Review by akrabat — APPROVED** [review]

**Comment by akrabat** [comment]
Thanks!

**Comment by akrabat** [comment]
This has been released as version [1.2.1](https://github.com/slimphp/Slim-Csrf/releases/tag/1.2.1)

**Pull request merged**